Makefile

```diff
@@ -190,27 +190,26 @@ unit-sup: build-launcher-for-supervisor-tests
 .PHONY: build-launcher-for-supervisor-tests
 
 # Lints we need to work through and decide as a team whether to allow or fix
-UNEXAMINED_LINTS = clippy::cyclomatic_complexity \
-                   clippy::large_enum_variant \
-                   clippy::len_without_is_empty \
-                   clippy::module_inception \
-                   clippy::needless_pass_by_value \
-                   clippy::needless_return \
-                   clippy::new_ret_no_self \
-                   clippy::new_without_default \
-                   clippy::new_without_default_derive \
-                   clippy::question_mark \
-                   clippy::redundant_field_names \
-                   clippy::too_many_arguments \
-                   clippy::trivially_copy_pass_by_ref \
-                   clippy::wrong_self_convention \
-                   renamed_and_removed_lints
+UNEXAMINED_LINTS =
 
 # Lints we disagree with and choose to keep in our code with no warning
-ALLOWED_LINTS =
+ALLOWED_LINTS = clippy::module_inception \
+                clippy::needless_return \
```

Comment:
I like having the option to have an explicit `return` in those large code blocks, but it would be nice to catch them on smaller ones, since that does strike me as noisy.

It might be worth enabling globally, but specifically excepting it on thorny functions where it is useful.

Further down the road, we might be able to use code that runs afoul of this as a signal that it could be refactored to something that's easier to read.

Reply:
Currently we have ~50 instances of this. If you're down to evaluate them on a case-by-case basis, I'm cool with moving this to `LINTS_TO_FIX`.

Reply:
I agree with enabling it and then just adding exceptions for those large code blocks. Those huge blocks are generally pretty hard to follow anyway, so it makes sense to me to use this as a signal to do some future refactoring.

Reply:
@baumanj Yeah, I'll be happy to evaluate our existing instances of this lint.

Reply:
+1 on this one.  Most situations where I've run into this recently have led to cleaner code on early return removal